Removes SBT from bin scripts

codefromthecrypt opened the pull request:
This ports configuration in existing bin scripts from SBT to gradle. It
also removes the redundant `-redis` scripts and renames `test` to
`tracegen`.

This also configures and explains why query and collector need to start
in the same directory by default (configKey == "dev").

jamescway commented on bin/collector:
```diff
@@ -1,6 +1,2 @@
-SERVICE_TYPE=${1:-$DEFAULT_TYPE}
-
-bin/sbt 'project zipkin-collector-service' "run -f zipkin-collector-service/config/collector-${SERVICE_TYPE}.scala"
+SERVICE_TYPE=${1:-dev} ./gradlew :zipkin-collector-service:run
```
is it still possible to run https://github.com/openzipkin/zipkin/blob/master/zipkin-collector-service/config/collector-cassandra.scala ?

codefromthecrypt replied:
yep syntax is `bin/collector cassandra`

blaines replied:
Hey there I was looking at this pull request with James and I had some comments. Everything looks to be working here, I just noticed the "dev" default set in multiple places. What can be done to improve this a bit is to use the gradle.properties file to set the `serviceType` default. That will eliminate the need to grab system environment variables. Also another thing that can be done is to use `-P` which sets a project property that overrides the default. So you could end up with something more like:
`./gradlew :zipkin-collector-service:run -PserviceType=${1}`
(in this case it may be needed to have a default defined here as well, `-PserviceType=${1:-dev}` so as to not accidentally nullify the value, but I didn't test myself)

```
gradle --help
-P, --project-prop      Set project property for the build script (e.g. -Pmyprop=myvalue).
```

codefromthecrypt replied:
Good point. Will use gradle.properties to default and -P to override.
Nullification concern from bash is still there, but at least the change is
more idiomatic. Thanks!

codefromthecrypt commented:
Don't merge, yet.

web is not working correctly via gradle. It seems to be missing something as the response is empty. The old script worked.

``` bash
$ curl -v http://localhost:8080/
* Hostname was NOT found in DNS cache
*   Trying ::1...
* Connected to localhost (::1) port 8080 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.37.1
> Host: localhost:8080
> Accept: */*
> 
* Empty reply from server
* Connection #0 to host localhost left intact
curl: (52) Empty reply from server
```

codefromthecrypt commented:
@blaines @jamescway PTAL I migrated the base config into gradle.properties. I think the default of "dev" in the two shell scripts is benign enough to leave alone.

I'm still looking into what's wrong with web. Also, sqlite broke some time ago.. I'll try to fix it in this change.

Commit:
Removes SBT from bin scripts

This ports configuration in existing bin scripts from SBT to gradle. It
also removes the redundant `-redis` scripts and renames `test` to
`tracegen`.

This also configures and explains why query and collector need to start
in the same directory by default (configKey == "dev").

codefromthecrypt commented:
This won't work until #522 is merged. Still looking into what's up with the web script in gradle.

codefromthecrypt commented on zipkin-collector-service/build.gradle:
```diff
@@ -1,6 +1,13 @@
 
+run {
+    // When configKey is dev, collector and query share ../zipkin.db
+    workingDir rootProject.projectDir
```
ps. this is the 'ah-hah!' which was screwing up collector vs query. Basically each was creating their own DB files.

rtyler replied:
I suggest changing this to `rootProject.buildDir` that way the generated DB file will be placed in `build/` and naturally ignored by git and cleaned up by `./gradlew clean`

codefromthecrypt replied:
SGTM

codefromthecrypt replied:
so the root build directory isn't created by default.. I'll have to make sure it is.

codefromthecrypt replied:
best way out was to point query to collector's build dir when configKey == "dev". The fundamental relationship is the coupling of these two processes, not to the root project. Ends up less work, too.

jamescway commented:
:+1:

abesto commented:
Cool! :+1:

rtyler commented:
one suggestion but mostly looks fine to me

codefromthecrypt commented:
yay.. works now! PTAL

codefromthecrypt commented on zipkin-web/build.gradle:
```diff
@@ -2,6 +2,11 @@ apply plugin: 'application'
 mainClassName = 'com.twitter.zipkin.web.Main'
 
+run {
+    args "-zipkin.web.resourcesRoot=${projectDir}/src/main/resources"
```
I bet @dsyer loves the fact we have to do this :)

codefromthecrypt commented:
Note: I've changed the gradle run config to always start in the build dir. Once we do things like actually creating log files, this should help reduce junk creation.

jamescway commented:
:+1: :+1:
